src/metabase/notification/send.clj

```diff
@@ -181,23 +181,87 @@
       (prometheus/observe! :metabase-notification/total-duration-ms {:payload-type payload_type} total-time))
     nil))
 
+(defn- cron->next-execution-times
+  "Returns the next n fired times for a given cron schedule."
+  [cron-schedule n]
+  (let [cron-expression (CronExpression. ^String cron-schedule)
+        now             (t/java-date)]
+    (loop [times     []
+           next-time now
+           remaining n]
+      (if (zero? remaining)
+        times
+        (let [next-execution (.getNextValidTimeAfter cron-expression next-time)]
+          (recur (conj times (t/instant next-execution))
+                 next-execution
+                 (dec remaining)))))))
+
+(defn- avg-interval-seconds
+  "Returns the average seconds between executions for a given cron schedule by sampling future execution times."
+  [cron-schedule n]
+  (assert (>= n 2) "Need at least 2 execution times to calculate average")
```

Comment:
What if we add support for "once off" notifications?

Reply:
forgot to mention but I've done this here: [a3f699d](https://github.com/metabase/metabase/pull/54647/commits/a3f699dcd8ae1cac7923bffb04b3180627965ab5)